internal/jitsi/jicofo.go

```diff
@@ -30,6 +39,57 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
+const JicofoName = "jicofo"
+
+type Jicofo struct {
+	client.Client
+	*v1beta1.Jicofo
+
+	ctx             context.Context
+	log             logr.Logger
+	name, namespace string
+	labels          map[string]string
+}
+
+func NewJicofo(ctx context.Context, c client.Client, l logr.Logger, req ctrl.Request) (Jitsi, error) {
+	j := &v1beta1.Jicofo{}
+	if err := c.Get(ctx, req.NamespacedName, j); err != nil {
+		return nil, err
+	}
+	defaultLabels := utils.GetDefaultLabels(JicofoName)
+	if !j.DeletionTimestamp.IsZero() {
+		return &Jicofo{
+			Client:    c,
+			Jicofo:    j,
+			name:      JicofoName,
+			namespace: j.Namespace,
+			ctx:       ctx,
+			log:       l,
+			labels:    defaultLabels,
+		}, meeterr.UnderDeletion()
+	}
+	if err := addFinalizerToJicofo(ctx, c, j); err != nil {
```

Comment:
consider [controller-runtime helpers](https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/controller/controllerutil#AddFinalizer) to handle finalizers

Reply:
Here is a [full example](https://sdk.operatorframework.io/docs/building-operators/golang/advanced-topics/#external-resources) from operator-sdk guide

Reply:
done